src/mesh.rs

```diff
@@ -376,195 +185,63 @@ impl<'a> Primitive<'a> {
-        self.find_accessor_with_semantic(Semantic::TexCoords(set))
-            .map(|accessor| TexCoords::from_accessor(accessor))
-    }
-
-    /// Returns the vertex texture co-ordinates of the given set, coerced into
-    /// `[f32; 2]` values.
-    pub fn tex_coords_f32(&self, set: u32) -> Option<TexCoordsF32> {
-        self.tex_coords(set).map(TexCoordsF32)
-    }
-
-    /// Returns the joint indices of the given set.
-    pub fn joints(&self, set: u32) -> Option<Joints> {
-        self.find_accessor_with_semantic(Semantic::Joints(set))
-            .map(|accessor| Joints::from_accessor(accessor))
-    }
-    
-    /// Returns the joint weights of the given set.
-    pub fn weights(&self, set: u32) -> Option<Weights> {
-        self.find_accessor_with_semantic(Semantic::Weights(set))
-            .map(|accessor| Weights::from_accessor(accessor))
-    }
-
-    /// Returns the primitive indices.
-    pub fn indices(&self) -> Option<Indices> {
-        self.json.indices.as_ref().map(|index| {
-            let accessor = self.mesh.gltf
-                .accessors()
-                .nth(index.value())
-                .unwrap()
-                .loaded(self.source);
-            Indices::from_accessor(accessor)
-        })
-    }
-
-    /// Returns the primitive indices, coerced into `u32` values.
-    pub fn indices_u32(&self) -> Option<IndicesU32> {
-        self.indices().map(IndicesU32)
-    }
-
-    /// Returns the primitive positions.
-    pub fn positions(&self) -> Option<Positions> {
-        self.find_accessor_with_semantic(Semantic::Positions)
-            .map(|accessor| unsafe {
-                Positions(accessor.iter())
-            })
-    }
-
-    /// Returns the primitive normals.
-    pub fn normals(&self) -> Option<Normals> {
-        self.find_accessor_with_semantic(Semantic::Normals)
-            .map(|accessor| unsafe {
-                Normals(accessor.iter())
-            })
-    }
-
-    /// Returns the primitive tangents.
-    pub fn tangents(&self) -> Option<Tangents> {
-        self.find_accessor_with_semantic(Semantic::Tangents)
-            .map(|accessor| unsafe {
-                Tangents(accessor.iter())
-            })
-    }
-
-    /// Returns the attribute with the given semantic value.
-    fn find_accessor_with_semantic(
-        &self,
-        semantic: Semantic,
-    ) -> Option<Loaded<'a, accessor::Accessor<'a>>> {
+    /// Return the accessor with the given semantic.
+    pub fn get(&self, semantic: &Semantic) -> Option<Accessor> {
         self.json.attributes
             .iter()
-            .find(|&(ref key, _)| key.as_ref().unwrap() == &semantic)
-            .map(|(_, index)| {
-                self.mesh.gltf
-                    .accessors()
-                    .nth(index.value())
-                    .unwrap()
-                    .loaded(self.source)
-            })
+            .find(|&(key, _)| key.as_ref().unwrap() == semantic)
```

Comment:
Using `get` means checking against a `&Checked<Sematic>` rather than a plain `&Semantic`. That's the only reason.